src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceInt_addPersonalCareNeedTest.kt

```diff
@@ -88,5 +89,29 @@ class BookingResourceInt_addPersonalCareNeedTest : ResourceTest() {
         .jsonPath("personalCareNeeds[1].endDate").isEqualTo(null)
         .jsonPath("personalCareNeeds[1].problemStatus").isEqualTo("ON")
     }
+
+    @Test
+    fun `will check for missing inputs values`() {
+      webTestClient.post()
+        .uri("/api/bookings/-1/personal-care-needs")
+        .headers(setAuthorisation(listOf("ROLE_MAINTAIN_HEALTH_PROBLEMS")))
+        .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+        .accept(MediaType.APPLICATION_JSON)
+        .body(
+          BodyInserters.fromValue(
+            """
+                {}
+            """.trimIndent()
+          )
+        )
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody()
+        .jsonPath("userMessage",
+          Matchers.any().equals("Field: commentText - must not be null"),
+          Matchers.any().equals("Field: startDate - must not be null"),
+          Matchers.any().equals("Field: problemStatus - must not be null"),
+          Matchers.any().equals("Field: problemCode - must not be null"))
+    }
```

Comment:
the userMessage was constructed of the various fields but the fields were generated randomly. This caused test failures between different test runs. So tested the fields individually